validator/project_validator.go

```diff
@@ -668,7 +707,7 @@ func ensureReferentialIntegrity(project *model.Project, distroIDs []string, dist
 			}
 
 			for _, distro := range task.RunOn {
-				if !utility.StringSliceContains(distroIDs, distro) && !utility.StringSliceContains(distroAliases, distro) {
+				if !utility.StringSliceContains(distroIDs, distro) && !utility.StringSliceContains(distroAliases, distro) && !containerNameMap[distro] {
 					errs = append(errs,
 						ValidationError{
 							Message: fmt.Sprintf("task '%s' in buildvariant '%s' "+
```

Comment:
"task '%s' in buildvariant '%s' references a nonexistent distro or container named '%s'"